### PR TITLE
Remove useless code

### LINKED ISSUE #18: Remove useless code
Remove useless code from Algolia
ex: all about replica, tag...

Should be done once all the tests are added. See #17

### PR DESCRIPTION
**Pull request by CaroFG**
Closes #18

### CODE REVIEW

File: lib/meilisearch/configuration.rb [hunk 1]
```diff
@@ -1,15 +1,15 @@
 module MeiliSearch
   module Configuration
     def configuration
-      @@configuration || raise(NotConfigured, "Please configure MeiliSearch. Set MeiliSearch.configuration = {application_id: 'YOUR_APPLICATION_ID', api_key: 'YOUR_API_KEY'}")
+      @@configuration || raise(NotConfigured, "Please configure MeiliSearch. Set MeiliSearch.configuration = {meilisearch_host: 'YOUR_meilisearch_host', meilisearch_api_key: 'YOUR_API_KEY'}")
```
Comment (curquiza) [inline lib/meilisearch/configuration.rb:4]:
```suggestion
      @@configuration || raise(NotConfigured, "Please configure MeiliSearch. Set MeiliSearch.configuration = {meilisearch_host: 'YOUR_MEILISEARCH_HOST', meilisearch_api_key: 'YOUR_API_KEY'}")
```

File: meilisearch-rails.gemspec [hunk 2]
```diff
@@ -79,19 +53,18 @@ Gem::Specification.new do |s|
 
     if Gem::Version.new(Gem::VERSION) >= Gem::Version.new('1.2.0') then
       s.add_runtime_dependency(%q<json>, [">= 1.5.1"])
-      s.add_runtime_dependency(%q<meilisearch>)
+      s.add_runtime_dependency(%q<meilisearch>, [">= 0.15.2"])
```
Comment (curquiza) [inline meilisearch-rails.gemspec:56]: Is it mandatory to give the version?

**Review by curquiza — CHANGES_REQUESTED** [review]
Really clean 👌  well done

File: meilisearch-rails.gemspec (hunk 2 above)
Comment (CaroFG) [reply meilisearch-rails.gemspec:56]: The second argument corresponds to the requirements of the given gem, so I'd say it's better if we keep it? https://docs.ruby-lang.org/en/2.6.0/Gem/Specification.html#method-i-add_runtime_dependency

Comment (curquiza) [reply meilisearch-rails.gemspec:56]: I'm afraid we use an obsolete version of the gem in our tests... the best would have the version of the one used by meilisearch-rails.

Comment (curquiza) [reply meilisearch-rails.gemspec:56]: I would rather have the last version than an obsolete version running in the tests.

Comment (CaroFG) [reply meilisearch-rails.gemspec:56]: Oh, so should I update it to [">= 0.15.**3**"]? Is that what you mean?

Comment (curquiza) [reply meilisearch-rails.gemspec:56]: Checked with @CaroFG, it's ok for me

**Review by curquiza — CHANGES_REQUESTED** [review]
We need to remove the useless parts in gemfile :)

File: Gemfile [hunk 3]
```diff
@@ -11,23 +11,6 @@ group :test do
...
-  elsif defined?(RUBY_VERSION) && RUBY_VERSION == "1.9.3"
-    gem 'rack', '< 2'
-    gem 'rack-cache', '<= 1.7.1'
-    if Gem::Version.new(ENV['RAILS_VERSION'] || '3.2.0') >= Gem::Version.new('4.0')
-      gem 'mime-types', '~> 2.6'
-    else
-      gem 'mime-types', '< 3'
-    end
-  end
   if defined?(RUBY_VERSION) &&
```
Comment (curquiza) [inline Gemfile:14]: This part is not useful if I'm correct

File: Gemfile [hunk 4]
```diff
@@ -59,9 +36,9 @@ end
 group :test, :development do
   gem 'will_paginate', '>= 2.3.15'
   if defined?(RUBY_VERSION) &&
-     defined?(RUBY_ENGINE) &&
-     RUBY_ENGINE == 'ruby' &&
-     Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2')
+    defined?(RUBY_ENGINE) &&
+    RUBY_ENGINE == 'ruby' &&
+    Gem::Version.new(RUBY_VERSION) < Gem::Version.new('2')
```
Comment (curquiza) [inline Gemfile:41]: I think we would remove this one? We don't support ruby version under 2.6

**Review by curquiza — COMMENTED** [review]
I think you can remove the `development` group in the Gemfile and also the gem `redgree`. Looks like redgree is not a maintained gem.

(sorry not to have noticed it at the same time)

**Comment by CaroFG** [comment]
> I think you can remove the `development` group in the Gemfile and also the gem `redgree`. Looks like redgree is not a maintained gem.
> 
> (sorry not to have noticed it at the same time)
Should I add the `dotenv` gem to the test group?

**Review by curquiza — APPROVED** [review]
🚀
Well don!!!

**Comment by CaroFG** [comment]
bors merge

**Pull request merged**